app/src/main/java/com/dessalines/thumbkey/ui/components/keyboard/KeyboardScreen.kt

```diff
@@ -296,6 +302,12 @@ fun KeyboardScreen(
             }
         }
     } else {
+        if (ctx.currentInputConnection.requestCursorUpdates(CURSOR_UPDATE_MONITOR or CURSOR_UPDATE_FILTER_INSERTION_MARKER)) {
```

Comment:
This line always fails, no matter what, and I can't figure out why.

Reply:
I didn't try it after adding the logging check, maybe that's the issue?

Reply:
It wouldn't make sense for that line to cause problems and not ctx.currentInputConnection.requestCursorUpdates(0) in the emoji section

Reply:
I don't think so... I haven't been able to get this to work at all tho.

Reply:
Idk what to tell you, chances are you know more about the quirks of android studio than I do. To build I go Build->Build Bundle(s)/APK(s)->Build APK(s), is that what you do? I also had API 34 downloaded. I'll try 21

Reply:
What sdk version are you using? I'm having problems with 21

Reply:
You generally add your dev device on the Device Manager tab on the right side of the screen, and it walks you through turning on wireless debugging, and deploying to your device from withing android-studio.

The build APK should work fine with the build.gradle minSdk versions, otherwise it would throw errors.

The main issue (that iirc I ran into before), is that `requestCursorUpdates` doesn't seem to work at all.

Reply:
What do you mean it doesn't work? It doesn't build? It doesn't lead to cursor updates being provided?

If the latter, what text field are you using it in? It depends on the text field to implement [updateCursorAnchorInfo](https://developer.android.com/reference/android/view/inputmethod/InputMethodManager#updateCursorAnchorInfo(android.view.View,%20android.view.inputmethod.CursorAnchorInfo)), so it doesn't work in e.g. the thumbkey text field, but it works for me in signal note to self/message compose

Reply:
I've just been copying `./app/build/outputs/apk/debug/app-debug.apk` to my phone and installing it 😅

Reply:
Does it always return false for you? from https://developer.android.com/reference/android/view/inputmethod/InputConnection#requestCursorUpdates(int)

> [returns] true if the request is scheduled. false to indicate that when the application will not call [InputMethodManager#updateCursorAnchorInfo( android.view.View, CursorAnchorInfo)](https://developer.android.com/reference/android/view/inputmethod/InputMethodManager#updateCursorAnchorInfo(android.view.View,%20android.view.inputmethod.CursorAnchorInfo)). Since Android [Build.VERSION_CODES.N](https://developer.android.com/reference/android/os/Build.VERSION_CODES#N) until [Build.VERSION_CODES.TIRAMISU](https://developer.android.com/reference/android/os/Build.VERSION_CODES#TIRAMISU), this API returned false when the target application does not implement this method.

Reply:
> It doesn't lead to cursor updates being provided?

Yep. 

Yes I was using the thumbkey text field to test. I'll try another field soon.

Reply:
I tried several programs, they all give the same result. We need to figure out another way to handle this one then.

Reply:
Did you try signal? But if it depends on niche functionality that most text fields don't implement, then I agree, although maybe it makes sense to do this plus other solutions?

I saw there's a function that gets x characters before the cursor, but it has a warning that it's slow and can lead to race conditions. But that might be the best bet, if lastaction==currentaction && preceedingchar == key.center.char

Reply:
Okay, I've pushed up some changes, that make it *partially* functional.

The problem is, it doesn't seem to work correctly with *any* jetpack compose text field, which is a huge problem.

Reply:
I've asked this on stackoverflow to see if anyone can help: https://stackoverflow.com/questions/77803840/requestcursorupdates-fails-without-logging-on-every-jetpack-compose-textfield